src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
+   *
+   * Setting the video tag dimension values only works with values in pixels.
+   * Percent values will not work.
+   * Some percents can be used, but width()/height() will return the number + %,
+   * not the actual computed width/height.
+   *
+   * @param  {Number|String=} num     New component height
+   * @param  {Boolean=} skipListeners Skip the resize event trigger
+   * @return {vjs.Component} This component, when setting the height
+   * @return {Number|String} The height, when getting
+   */
+  height(num, skipListeners){
+    return this.dimension('height', num, skipListeners);
+  }
 
-/**
- * Remove a CSS class name from the component's element
- *
- * @param {String} classToRemove Classname to remove
- * @return {vjs.Component}
- */
-Component.prototype.removeClass = function(classToRemove){
-  Lib.removeClass(this.el_, classToRemove);
-  return this;
-};
+  /**
+   * Set both width and height at the same time
+   *
+   * @param  {Number|String} width
+   * @param  {Number|String} height
+   * @return {vjs.Component} The component
+   */
+  dimensions(width, height){
+    // Skip resize listeners on width for optimization
+    return this.width(width, true).height(height);
+  }
 
-/**
- * Show the component element if hidden
- *
- * @return {vjs.Component}
- */
-Component.prototype.show = function(){
-  this.removeClass('vjs-hidden');
-  return this;
-};
+  /**
+   * Get or set width or height
+   *
+   * This is the shared code for the width() and height() methods.
+   * All for an integer, integer + 'px' or integer + '%';
+   *
+   * Known issue: Hidden elements officially have a width of 0. We're defaulting
+   * to the style.width value and falling back to computedStyle which has the
+   * hidden element issue. Info, but probably not an efficient fix:
+   * http://www.foliotek.com/devblog/getting-the-width-of-a-hidden-element-with-jquery-using-width/
+   *
+   * @param  {String} widthOrHeight  'width' or 'height'
+   * @param  {Number|String=} num     New dimension
+   * @param  {Boolean=} skipListeners Skip resize event trigger
+   * @return {vjs.Component} The component if a dimension was set
+   * @return {Number|String} The dimension if nothing was set
+   * @private
+   */
+  dimension(widthOrHeight, num, skipListeners){
+    if (num !== undefined) {
+      // Set to zero if null or literally NaN (NaN !== NaN)
+      if (num === null || num !== num) {
+        num = 0;
+      }
 
-/**
- * Hide the component element if currently showing
- *
- * @return {vjs.Component}
- */
-Component.prototype.hide = function(){
-  this.addClass('vjs-hidden');
-  return this;
-};
+      // Check if using css width/height (% or px) and adjust
+      if ((''+num).indexOf('%') !== -1 || (''+num).indexOf('px') !== -1) {
+        this.el_.style[widthOrHeight] = num;
+      } else if (num === 'auto') {
+        this.el_.style[widthOrHeight] = '';
+      } else {
+        this.el_.style[widthOrHeight] = num+'px';
```

Comment:
Yeah, not saying that it's an issue. Just that we should be careful around things like these because `num` is user inputtable and if it's carefully crafted, could cause issues with template strings since they have access to the variables.